components/brave_new_tab_ui/components/search/config.ts

```diff
@@ -26,38 +22,57 @@ const getConfig = () => {
   return cache!
 }
 
-// Determines whether any search engines are enabled
-export const hasEnabledEngine = () => Object.values(getConfig()).find(c => c)
+export function useNtpWidgetDefaultSearchEngine() {
+  const [lastUsed, setLastUsed] = useNewTabPref('lastUsedNtpSearchEngine')
+  const lastLocalStorage = localStorage.getItem(LAST_SEARCH_ENGINE_KEY)
+  const config = getEnabledSearchEngines()
 
-export const maybeEnableDefaultEngine = () => {
-  // If no engines are enabled, enable the default one.
-  if (!hasEnabledEngine()) {
-    setEngineEnabled({ host: braveSearchHost }, true)
-  }
-}
+  const setLastNtpSearchEngine = React.useCallback((engine: SearchEngineInfo) => {
+    setLastUsed(engine.host);
+    localStorage.setItem(LAST_SEARCH_ENGINE_KEY, engine.host);
+  }, [setLastUsed])
 
-export const setEngineEnabled = (engine: { host: string }, enabled: boolean) => {
-  const config = getConfig()
-  config[engine.host] = enabled
+  const lastNtpSearchEngine = React.useCallback(() => {
+    let last = lastUsed;
 
-  localStorage.setItem(ENABLED_SEARCH_ENGINES_KEY, JSON.stringify(config))
-}
+    if (lastLocalStorage && config[lastLocalStorage] && lastUsed !== lastLocalStorage) {
+      setLastUsed(lastLocalStorage);
+      last = lastLocalStorage;
```

Comment:
should we delete the value from `localStorage` at this point?


```suggestion
      last = lastLocalStorage;
      localStorage.removeItem(LAST_SEARCH_ENGINE_KEY)
```

Reply:
I think we should also comment that this is a migration and will be run at most once